src/xcode/ENA/ENA/Source/Services/DSCList/DSCListProvider.swift

```diff
@@ -0,0 +1,87 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import UIKit
+import OpenCombine
+import HealthCertificateToolkit
+
+final class DSCListProvider: DSCListProviding {
+
+	// MARK: - Init
+
+	init(
+		client: DSCListFetching,
+		store: Store,
+		interval: TimeInterval = 12 * 60 * 60  // 12 hours
+	) {
+		self.client = client
+		self.store = store
+		self.interval = interval
+		// load last knows | default DSCList data
+		self.metaData = Self.loadDSCListMetaDataIfAvailable(store: store, interval: interval)
+		self.signingCertificates = CurrentValueSubject<[DCCSigningCertificate], Never>(metaData.signingCertificate)
+
+		// trigger an update immediately
+		NotificationCenter.default.addObserver(self, selector: #selector(updateListIfNeeded), name: UIApplication.willEnterForegroundNotification, object: nil)
+		updateListIfNeeded()
+	}
+
+	deinit {
+		NotificationCenter.default.removeObserver(self, name: UIApplication.willEnterForegroundNotification, object: nil)
+	}
+
+	// MARK: - Protocol DSCListProviding
+
+	private(set) var signingCertificates: CurrentValueSubject<[DCCSigningCertificate], Never>
+
+	// MARK: - Internal
+
+	private(set) var metaData: DSCListMetaData {
+		// on change write to the store and update publisher
+		didSet {
+			store.dscList = metaData
+			signingCertificates.value = metaData.signingCertificate
+		}
+	}
+
+	// MARK: - Private
+
+	private let client: DSCListFetching
+	private let store: Store
+	private let interval: TimeInterval
+
+	private static func loadDSCListMetaDataIfAvailable(store: Store, interval: TimeInterval) -> DSCListMetaData {
+		guard let metaDataDSCList = store.dscList else {
+			// store is empty -> store default and return it
+			guard
+				let url = Bundle.main.url(forResource: "default_dsc_list", withExtension: "bin"),
+				let data = try? Data(contentsOf: url),
+				let dscList = try? SAP_Internal_Dgc_DscList(serializedData: data)
+			else {
+				fatalError("Failed to read default DSCList bin file - set empty fallback")
```

Comment:
I think fatalError could be avoided here with a simple log and return empty object.

Reply:
The FatalError is here for a reason - if the fallback value - the file - was not found it is critical and will break the DSCListProvider. That's why I added a FatalError. Also there is a Unit-test making sure the file is found in the bundle - so this one shall never happen.